compiler/parser/chapel.ypp

```diff
@@ -1367,6 +1368,56 @@ opt_init_expr:
 | TASSIGN opt_try_expr  { $$ = $2; }
 ;
 
+
+ret_array_type:
+  TLSBR TRSBR type_level_expr
+    { $$ = new CallExpr("chpl__buildArrayRuntimeType", gNil, $3); }
+| TLSBR TRSBR
+    { $$ = new CallExpr("chpl__buildArrayRuntimeType", gNil, NULL); }
+| TLSBR expr_ls TRSBR type_level_expr
+    { $$ = new CallExpr("chpl__buildArrayRuntimeType",
+             new CallExpr("chpl__ensureDomainExpr", $2), $4);
+    }
+| TLSBR expr_ls TRSBR
+    { $$ = new CallExpr("chpl__buildArrayRuntimeType",
+             new CallExpr("chpl__ensureDomainExpr", $2), NULL);
+    }
+| TLSBR TRSBR ret_array_type
+    { $$ = new CallExpr("chpl__buildArrayRuntimeType", gNil, $3); }
+| TLSBR expr_ls TRSBR ret_array_type
+    { $$ = new CallExpr("chpl__buildArrayRuntimeType",
+             new CallExpr("chpl__ensureDomainExpr", $2), $4);
+    }
+| TLSBR expr_ls TIN expr TRSBR type_level_expr
```

Comment:
I thought the PR message said this case wasn't working due to parser conflicts?

Reply:
Nah, this is the case that "works".  The parser barked when the `type_level_expr` part was elided.  That said, I didn't add a test for this, since it was functionality that got grandfathered over from when we were using the same array type stuff as variable declarations

Reply:
I'd prefer we either leave this case out of the parser or add a test for it, if you havn't already